### ISSUE TITLE
Unable to reproduce adaptation results

### ISSUE THREAD

**Issue by vishaal27** [facebookresearch/SIMAT#4]
Hi,

Congrats on the amazing work!

I am trying to reproduce the results of your fine-tuning CLIP on MS-COCO experiment (figure 5, section 5.2 in the [paper](https://openaccess.thecvf.com/content/CVPR2022W/ODRUM/papers/Couairon_Embedding_Arithmetic_of_Multimodal_Queries_for_Image_Retrieval_CVPRW_2022_paper.pdf)) -- however I am running into issues while doing the exact fine-tuning and am getting lower SIMAT scores than reported in your paper.

These are the exact training loss and validation loss plots while doing the fine-tuning:

Adaption at tau=0.01:

![image](https://user-images.githubusercontent.com/21314772/177142727-acd28a93-c372-4e46-b22b-7a8d2f7c26f5.png)

Adaption at tau=0.1:

![image](https://user-images.githubusercontent.com/21314772/177142806-86f35745-a8b9-4562-a752-e8e8eb74db77.png)

I used the exact same hyperparameter settings as you have done in your adaptation script:

-  learning rate=1e-3 
-  lr decay schedule with step_size=25 and gamma=0.1
-  num_epochs=50
-  gradient clipping at norm=1

Do you have any insights into where the training script might be going wrong and why the loss seems to be stagnating as we step through training? After adaptation (with these training plots), I get a SIMAT score of 37.10 compared to your 47.5 (at tau=0.1, lambda=1). Similarly, I get a SIMAT score of around 16.61 compared to your 17.10 (at tau=0.01, lambda=1).

Note: I had to reimplement chunks of your code in pytorch as I believe parts of your adaptation script were incomplete in Pytorch lightning -- I would be happy to share my adaptation script with you if that would help!

Hoping for a prompt response.
